core/src/main/java/com/linecorp/armeria/common/logging/DefaultRequestLog.java

```diff
@@ -1811,6 +1663,13 @@ public List<RequestLogAccess> children() {
             return DefaultRequestLog.this.children();
         }
 
+        @Override
+        public Set<RequestLogProperty> availableProperties() {
+            return Arrays.stream(RequestLogProperty.values())
+                         .filter(this::isAvailable)
+                         .collect(Collectors.toSet());
```

Comment:
Could we make sure to return an immutable `EnumSet`? You could use Guava's `Sets.toImmutableEnumSet()`.